src/pykiso/global_config.py

```diff
@@ -135,15 +140,21 @@ def grab_cli_config(func) -> Callable:
         """
 
         @functools.wraps(func)
-        def grab_inner(*args, **kwargs) -> None:
+        def grab_inner(click_context: click.Context, *args, **kwargs) -> None:
             """Grab the given values from cli entry point level and
             store it in the GlobalConfig instance.
 
+            :param click_context: click context
             :param args: positonal arguments
             :param kwargs: named arguments
             """
-            object_config = Grabber.create_config_object(kwargs)
+            kwargs_copy = copy(kwargs)
+            click_args = cli.eval_user_tags(click_context)
+            for tag in list(click_args.keys()):
+                click_args[tag.replace("-", "_")] = click_args.pop(tag)
+            kwargs_copy.update(click_args)
```

Comment:
You could also avoid the dict copy by merging both dicts into a new one:
`kwargs_with_tags = {**click_args, **kwargs}`

Reply:
Done